src/models/git/FileStatus.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Represent a git branch.
+ */
+import { FileInformation } from 'leto-modelizer-plugin-core';
+
+class FileStatus extends FileInformation {
+  /**
+   * Default constructor.
+   * @param {String} filename - File name.
+   * @param {Number} headStatus - The HEAD status is either absent (0) or present (1).
+   * @param {Number} workdirStatus - The WORKDIR status is either absent (0), identical to HEAD (1),
+   * or different from HEAD (2).
+   * @param {Number} stageStatus - The STAGE status is either absent (0), identical to HEAD (1),
+   * identical to WORKDIR (2), or different from WORKDIR (3).
+   */
+  constructor(props = {
+    headStatus: -1,
+    workdirStatus: -1,
+    stageStatus: -1,
+  }) {
+    super(props);
+    this.matrix = `${props.headStatus}${props.workdirStatus}${props.stageStatus}`;
+    this.isNew = false;
+    this.isUntracked = false;
+    this.isAdded = false;
+    this.isStaged = false;
+    this.hasUnstagedChanged = false;
+    this.isDeleted = false;
+    this.isUnstaged = false;
+    this.isUnmodified = false;
+
+    switch (this.matrix) {
+      case '000': // The HEAD status is absent, the WORKDIR status is absent and the STAGE status is absent.
```

Comment:
too much `case`. Can be reduce. Useless comments (thx to the good function description)

Reply:
done